pkg/controller/provider/netlify/access.go

```diff
@@ -99,57 +100,31 @@ func (this *access) CreateRecord(r raw.Record) error {
 	}
 	this.metrics.AddRequests(provider.M_CREATERECORDS, 1)
 	this.rateLimiter.Accept()
-	params := operations.CreateDNSRecordParams{
-		ZoneID: a.DNSZoneID,
-		DNSRecord: &dnsRecord,
-	}
-	_, err := this.client.CreateDNSRecord(&params, this.authInfo)
+	createParams := operations.NewCreateDNSRecordParams()
+	createParams.SetZoneID(a.DNSZoneID)
+	createParams.SetDNSRecord(&dnsRecord)
+	_, err := this.client.CreateDNSRecord(createParams, this.authInfo)
 	return err
 }
 
 func (this *access) UpdateRecord(r raw.Record) error {
-	a := r.(*Record)
-	ttl := r.GetTTL()
-	testTTL(&ttl)
 	// Netlify does not support updating a record
 	// Delete the existing record and re-create it
-	deleteParams := operations.DeleteDNSRecordParams{
-		ZoneID: a.DNSZoneID,
-		DNSRecordID: a.ID,
-	}
-	// Update records is not atomic as it's made of delete + create
-	// We update the metrics before the delete is started
-	this.metrics.AddRequests(provider.M_UPDATERECORDS, 1)
-	this.rateLimiter.Accept()
-	_, err := this.client.DeleteDNSRecord(&deleteParams, this.authInfo)
+	err := this.DeleteRecord(r)
```

Comment:
The reason I did not do this before is because the `DeleteRecord` and `CreateRecord` methods both update the metrics, and I wasn't sure if you wanted to have the update method matched by two metrics delete + create.
It really depends if the metrics are about counting the incoming requests or the outgoing ones. I'm guessing the latter?

Reply:
exactly, they should provide info about the usage of the external rest api